### PR TITLE
linking fix_error function to views and controller

### PR DESCRIPTION
**Pull request by jenachoi**

### CODE REVIEW

File: app/controllers/exam_templates_controller.rb [hunk 1]
```diff
@@ -157,6 +157,7 @@ def download_error_file
                         @assignment.short_identifier, 'error', params[:file_name]),
               filename: params[:file_name],
               type: 'application/pdf')
+    session[:filename] = params[:file_name]
```
Comment (david-yz-liu) [inline app/controllers/exam_templates_controller.rb]: Don't use the `session` hash.

File: app/controllers/exam_templates_controller.rb [hunk 2]
```diff
@@ -166,6 +167,15 @@ def download_error_file_path
                 show_in_browser: true )
   end
 
+  def fix_error
+    exam_template = ExamTemplate.find(params[:fix_error][:exam_template])
+    copy_number = params[:fix_error][:copy]
+    page_number = params[:fix_error][:page]
+    filename = session[:filename]
```
Comment (david-yz-liu) [inline app/controllers/exam_templates_controller.rb]: This should be sent in the `params`. You might need to create a hidden input.

File: app/views/exam_templates/assign_errors.html.erb [hunk 3]
```diff
@@ -207,21 +207,20 @@
         <div class="side_header">
           <h2><%= t('exam_templates.assign_errors.document_info') %></h2>
         </div>
-        <%= form_tag(nil, id: 'info_form', class:'side_header') do %>
+        <%= form_for :fix_error,
+                     url: fix_error_assignment_exam_templates_path(@assignment),
+                     html: { multipart: true, size: 1, class: 'side_header' } do |f| %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/assign_errors.html.erb]: I'm not sure why you keep using `multipart` for forms. It really shouldn't be necessary.

File: app/views/exam_templates/assign_errors.html.erb [hunk 4]
```diff
@@ -207,21 +207,20 @@
...
           </div>
-          <%= label_tag(:copy_number, t('exam_templates.assign_errors.copy_number')) %>
-          <%= text_field_tag(:copy_number, nil) %><br>
-          <%= label_tag(:page_number, t('exam_templates.assign_errors.page_number')) %>
-          <%= text_field_tag(:page_number, nil) %><br>
-          <%= label_tag(:upside_down,  t('exam_templates.assign_errors.upside_down')) %>
-          <%= check_box_tag(:upside_down, nil) %><br>
-          <br>
-          <%= submit_tag(t('save')) %>
+          <%= f.label(:copy, t('exam_templates.assign_errors.copy_number') + ':') %>
```
Comment (david-yz-liu) [inline app/views/exam_templates/assign_errors.html.erb]: Don't make changes I didn't ask you to do. Here, you inserted a colon.

File: app/views/exam_templates/assign_errors.html.erb [hunk 5]
```diff
@@ -207,21 +207,20 @@
...
+        <%= form_for :fix_error,
+                     url: fix_error_assignment_exam_templates_path(@assignment),
+                     html: { multipart: true, size: 1, class: 'side_header' } do |f| %>
           <div class="side_header">
-            <%= label_tag(:exam_template, @assignment.exam_templates.model_name.human + ':') %>
-            <%= select_tag(:exam_template,
-                           options_from_collection_for_select(@assignment.exam_templates.where(assignment_id: @assignment.id), :id, :name))
+            <%= f.label(:exam_template, ExamTemplate.model_name.human) %>
+            <%= f.select(:exam_template,
+                         options_from_collection_for_select(ExamTemplate.where(assignment_id: @assignment.id), :id, :name))
```
Comment (david-yz-liu) [inline app/views/exam_templates/assign_errors.html.erb]: Keep the association, but you can remove the `where`.

File: app/views/exam_templates/assign_errors.html.erb [hunk 6]
```diff
@@ -207,21 +207,20 @@
         <div class="side_header">
           <h2><%= t('exam_templates.assign_errors.document_info') %></h2>
         </div>
-        <%= form_tag(nil, id: 'info_form', class:'side_header') do %>
+        <%= form_for :fix_error,
```
Comment (david-yz-liu) [inline app/views/exam_templates/assign_errors.html.erb]: Switch to using `form_tag`.

**Comment by jenachoi** [comment]
still working on it!

**Pull request merged**